### PR TITLE
Causeway 3557

### PR DESCRIPTION
**Pull request by danhaywood**

### CODE REVIEW

File: commons/src/main/java/org/apache/causeway/commons/collections/Can.java [hunk 1]
```diff
@@ -176,29 +188,48 @@ public static <T> Can<T> of(final T ... array) {
...
+     * @param <T>
+     */
+    public static <T> Can<T> ofArray(T[] array, Function<T, T> nullMapper) {
         if(_NullSafe.size(array)==0) {
             return empty();
         }
 
         val nonNullElements = Stream.of(array)
                 .filter(_NullSafe::isPresent)
+                .map(nullMapper)
```
Comment (andi-huber) [inline commons/src/main/java/org/apache/causeway/commons/collections/Can.java:219]: needs to happen before the filter

File: commons/src/main/java/org/apache/causeway/commons/collections/Can.java [hunk 2]
```diff
@@ -176,29 +188,48 @@ public static <T> Can<T> of(final T ... array) {
...
+    }
 
+    /**
+     *
+     * @param array
+     * @param nullMapper -
+     * @return
+     * @param <T>
+     */
+    public static <T> Can<T> ofArray(T[] array, Function<T, T> nullMapper) {
```
Comment (andi-huber) [inline commons/src/main/java/org/apache/causeway/commons/collections/Can.java:212]: arguments: **array** is `@Nullable`  while **nullMapper** is `@NonNull`

File: commons/src/main/java/org/apache/causeway/commons/collections/Can.java [hunk 3]
```diff
@@ -59,8 +59,10 @@
  * <p>
  * Same idiomatic convention applies: References to {@link Can}
  * should never be initialized to {@code null}.
+ *
  * <p>
- * A {@link Can} must not contain elements equal to {@code null}.
+ * <b>IMPORTANT:</b>A {@link Can} must not contain elements equal to {@code null}.  If you want to store a null within
```
Comment (andi-huber) [inline commons/src/main/java/org/apache/causeway/commons/collections/Can.java:64]: I guess the second sentence does not really add any information.

Comment (andi-huber) [reply commons/src/main/java/org/apache/causeway/commons/collections/Can.java:64]: That said, `Can<T>` simply does not store `null`. Working around that fact is bad practice. Instead use a different container.

Comment (danhaywood) [reply commons/src/main/java/org/apache/causeway/commons/collections/Can.java:64]: updated in subsequent commit direct to `master`.

**Pull request merged**